qemu/tests/timerdevice_change_guest_clksource.py

```diff
@@ -41,134 +35,38 @@ def verify_guest_clock_source(session, expected):
-            else:
-                kernel_cfg_new = "%s %s" % (kernel_cfg,
-                                            "clocksource=%s" % clksrc)
-            disk_obj.replace_image_file_content(grub_file, kernel_cfg,
-                                                kernel_cfg_new)
-
-            error_context.context("Boot the guest", logging.info)
-            if clksrc != "kvm-clock":
-                cpu_model_flags = params.get("cpu_model_flags")
-                if "-kvmclock" not in cpu_model_flags:
-                    params["cpu_model_flags"] = cpu_model_flags + ",-kvmclock"
-            vm_name = params["main_vm"]
-            env_process.preprocess_vm(test, params, env, vm_name)
-            vm = env.get_vm(vm_name)
-            vm.verify_alive()
+            utils_time.update_clksrc(vm, clksrc=avl_clksrc)
             session = vm.wait_for_login(timeout=timeout)
-
             error_context.context("Check the current clocksource in guest",
                                   logging.info)
-            verify_guest_clock_source(session, clksrc)
+            verify_guest_clock_source(session, avl_clksrc)
     finally:
-        try:
-            error_context.context("Shutdown guest")
-            vm.destroy()
-            error_context.context("Restore guest kernel cli", logging.info)
-            image_filename = storage.get_image_filename(params,
-                                                        data_dir.get_data_dir())
-            grub_file = params.get("grub_file", "/boot/grub2/grub.cfg")
-            kernel_cfg_pattern = params.get("kernel_cfg_pos_reg",
-                                            r".*vmlinuz-\d+.*")
-
-            disk_obj = utils_disk.GuestFSModiDisk(image_filename)
-            kernel_cfg_original = disk_obj.read_file(grub_file)
-            try:
-                kernel_cfg = re.findall(kernel_cfg_pattern,
-                                        kernel_cfg_original)[0]
-            except IndexError as detail:
-                test.error("Couldn't find the kernel config, regex"
-                           " pattern is '%s', detail: '%s'" %
-                           (kernel_cfg_pattern, detail))
-
-            if "clocksource=" in kernel_cfg:
-                kernel_cfg_new = re.sub(
-                    r"clocksource=[a-z \-_]+", " ", kernel_cfg)
-                disk_obj.replace_image_file_content(grub_file, kernel_cfg,
-                                                    kernel_cfg_new)
-        except Exception as detail:
-            logging.error("Failed to restore guest kernel cli."
-                          " Detail: '%s'" % detail)
+        error_context.context("Restore guest kernel cli", logging.info)
+        proc_cmdline = "cat /proc/cmdline"
+        check_output = str(session.cmd(proc_cmdline, timeout=60))
+        clk_removed = re.search("clocksource.*", check_output).group()
+        utils_test.update_boot_option(vm, args_removed=clk_removed)
```

Comment:
Directly `utils_test.update_boot_option(vm, args_removed='clocksource')` should be sufficient.

Reply:
It isn't enough, because before removing param, the param will be checked if it exists.

Reply:
@YanhuiMa From the code: https://github.com/avocado-framework/avocado-vt/blob/master/virttest/utils_test/__init__.py#L151 I suppose no matter 'clocksource' or 'clocksource=sth' is passed, it should return correct result, isn't it?

Reply:
check_output = str(session.cmd(proc_cmdline, timeout=60)).split()
It is a list with "clocksource=xxx", if only use "clocksource" here, "if each_arg in check_output" will return false.